CMakeLists.txt

```diff
@@ -3573,6 +3574,26 @@ if(ENABLE_DASHBOARD)
   endif()
 endif()
 
+#
+# Ship CMakeCache.txt as an archive
+#
+
+file(ARCHIVE_CREATE
+    OUTPUT "${CMAKE_BINARY_DIR}/${BUILD_INFO_CMAKE_CACHE_ARCHIVE_NAME}"
+    PATHS "${CMAKE_BINARY_DIR}/CMakeCache.txt"
+    FORMAT raw
+    COMPRESSION GZip
+    COMPRESSION_LEVEL 9
+)
```

Comment:
This probably needs to happen at build time, not configure time. CMake does not appear to provide any guarantee that the cache file will exist on-disk until configuration is complete.